diff --git a/tests/test_webApi.py b/tests/test_webApi.py
index c73a755..0cc37fe 100644
--- a/tests/test_webApi.py
+++ b/tests/test_webApi.py
@@ -1,6 +1,7 @@
 
 import json
 from app import create_app
+import math
 
 app = create_app('test')
 
@@ -19,6 +20,16 @@ def test_NonPrank():
             "/nonlinear/personal_rank", json=json_data)
         assert response.status_code == 200
 
+        print(response.json)
+
+        for r in response.json['feedItemsRank']:
+          g = r.get('global')
+          p = r.get('personalised')
+          assert type(g) is float
+          assert not math.isnan(g)
+          assert type(p) is float
+          assert not math.isnan(p)
+
 
 def test_NonGrank():
     with app.test_client() as c:
@@ -28,3 +39,10 @@ def test_NonGrank():
         response = c.post(
             "/nonlinear/global_rank", json=json_data)
         assert response.status_code == 200
+
+        print(response.json)
+
+        for r in response.json['feedItemsRank']:
+          g = r.get('global')
+          assert type(g) is float
+          assert not math.isnan(g)
